api/v1alpha1/tcproute_types.go

```diff
@@ -16,44 +16,51 @@ limitations under the License.
 package v1alpha1
 
 import (
+	core "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// TcpRouteSpec defines the desired state of TcpRoute
-type TcpRouteSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+// TCPRouteSpec defines the desired state of the TCPRoute.
+type TCPRouteSpec struct {
+	// Hosts is a list of Host definitions.
+	Hosts []TCPRouteHost `json:"hosts,omitempty"`
 }
 
-// TcpRouteStatus defines the observed state of TcpRoute
-type TcpRouteStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+// TCPRouteHost is the configuration for a given host.
+type TCPRouteHost struct {
+	// Hostname is a fully qualified domain name.  This must be a registered
+	// name as defined by RFC 3986.  Note that IP addresses cannot be used.
+	Hostname string `json:"hostname"`
```

Comment:
There should be a destination for each host in this section, similar to `HTTPRouteAction` in the `HTTPRoute` resource. Am I missing something?

Reply:
Yes, we will need to add targets as well.  I wanted to focus this PR on the pieces that would enable the three use-cases that I listed (edge-termination, re-encrypt, and pass-through TLS), but I can fill out the `TCPRoute` target as part of this PR if that would make the idea clearer.

Reply:
I'd recommend to have at least a mock struct in place with a TODO.